src/charm.py

```diff
@@ -410,8 +406,14 @@ def _configure(self, event: Optional[ConfigChangedEvent] = None) -> None:  # noq
                 return
         except VaultClientError:
             return
-        if not all(self._get_approle_auth_secret()):
+
+        if not all(self._get_approle_auth_secret()) and not self.unit.is_leader():
             return
+        if not all(self._get_approle_auth_secret()) and self.unit.is_leader():
+            token = self._get_approle_temporary_token()
```

Comment:
Do we need to handle the expiry of the token and the expiry of the secret?

Reply:
changed the whole thing, please look again.